gtsam/hybrid/GaussianHybridFactorGraph.cpp

```diff
@@ -0,0 +1,166 @@
+/* ----------------------------------------------------------------------------
+ * Copyright 2021 The Ambitious Folks of the MRG
+ * See LICENSE for the license information
+ * -------------------------------------------------------------------------- */
+
+/**
+ * @file   GaussianHybridFactorGraph.cpp
+ * @brief  Custom hybrid factor graph for discrete + continuous factors
+ * @author Kevin Doherty, kdoherty@mit.edu
+ * @date   December 2021
+ */
+
+#include <gtsam/discrete/DiscreteEliminationTree.h>
+#include <gtsam/discrete/DiscreteJunctionTree.h>
+#include <gtsam/hybrid/DCGaussianMixtureFactor.h>
+#include <gtsam/hybrid/GaussianHybridFactorGraph.h>
+#include <gtsam/hybrid/HybridEliminationTree.h>
+#include <gtsam/inference/EliminateableFactorGraph-inst.h>
+#include <gtsam/linear/GaussianEliminationTree.h>
+#include <gtsam/linear/GaussianFactorGraph.h>
+#include <gtsam/linear/GaussianJunctionTree.h>
+#include <gtsam/linear/HessianFactor.h>
+
+#include <boost/make_shared.hpp>
+
+using namespace std;
+
+namespace gtsam {
+
+// Instantiate base classes
+template class EliminateableFactorGraph<GaussianHybridFactorGraph>;
+
+void GaussianHybridFactorGraph::print(
+    const string& str, const gtsam::KeyFormatter& keyFormatter) const {
+  Base::print(str, keyFormatter);
+  factorGraph_.print("GaussianGraph", keyFormatter);
+}
+
+DiscreteKeys GaussianHybridFactorGraph::discreteKeys() const {
```

Comment:
Why is that here? Expected in HFG...

Reply:
It's an override specifically for dealing with factors in `discreteGraph_` which are `DecisionTreeFactor`s.

Reply:
But that is there in HFG…

Reply:
I thought there was some difference when I split the code but no test fails so yep I'll remove this. :)